streams/src/test/java/org/apache/kafka/streams/processor/internals/StandbyTaskTest.java

```diff
@@ -213,6 +224,177 @@ public void testUpdate() throws IOException {
+        task.commit();
+
+        final List<ConsumerRecord<byte[], byte[]>> remaining2 = task.update(topicPartition, remaining1);
+
+        // the first record's window should have expired.
+        assertEquals(
+            Collections.singletonList(
+                new KeyValue<>(new Windowed<>(1, new TimeWindow(0, 60_000)), 100L)
+            ),
+            getStoreContents(storeName, task)
+        );
+
+        consumer.commitSync(Utils.mkMap(Utils.mkEntry(topicPartition, new OffsetAndMetadata(21L))));
+        task.commit();
+
+        final List<ConsumerRecord<byte[], byte[]>> remaining3 = task.update(topicPartition, remaining2);
+
+        // the first record's window should have expired.
+        assertEquals(
+            Arrays.asList(
+                new KeyValue<>(new Windowed<>(1, new TimeWindow(0, 60_000)), 100L),
+                new KeyValue<>(new Windowed<>(2, new TimeWindow(60_000, 120_000)), 100L)
+            ),
+            getStoreContents(storeName, task)
+        );
+
+        consumer.commitSync(Utils.mkMap(Utils.mkEntry(topicPartition, new OffsetAndMetadata(31L))));
+        task.commit();
+
+        final List<ConsumerRecord<byte[], byte[]>> remaining4 = task.update(topicPartition, remaining3);
+
+        // the first record's window should have expired.
+        assertEquals(
+            Arrays.asList(
+                new KeyValue<>(new Windowed<>(1, new TimeWindow(0, 60_000)), 100L),
+                new KeyValue<>(new Windowed<>(2, new TimeWindow(60_000, 120_000)), 100L),
+                new KeyValue<>(new Windowed<>(3, new TimeWindow(120_000, 180_000)), 100L)
+            ),
+            getStoreContents(storeName, task)
+        );
+
+
+        consumer.commitSync(Utils.mkMap(Utils.mkEntry(topicPartition, new OffsetAndMetadata(41L))));
+        task.commit();
+
+        final List<ConsumerRecord<byte[], byte[]>> remaining5 = task.update(topicPartition, remaining4);
+        assertNull(remaining5);
+
+        // the first record's window should have expired.
+        assertEquals(
+            Arrays.asList(
+                new KeyValue<>(new Windowed<>(2, new TimeWindow(60_000, 120_000)), 100L),
+                new KeyValue<>(new Windowed<>(3, new TimeWindow(120_000, 180_000)), 100L),
+                new KeyValue<>(new Windowed<>(4, new TimeWindow(180_000, 240_000)), 100L)
+            ),
+            getStoreContents(storeName, task)
+        );
+
+        restoreStateConsumer.commitSync();
+        task.commit();
+
+        task.closeStateManager(true);
+
+        File taskDir = stateDirectory.directoryForTask(taskId);
+        OffsetCheckpoint checkpoint = new OffsetCheckpoint(new File(taskDir, ProcessorStateManager.CHECKPOINT_FILE_NAME));
+        Map<TopicPartition, Long> offsets = checkpoint.read();
+
+        assertEquals(1, offsets.size());
+        assertEquals(new Long(41L), offsets.get(topicPartition));
```

Comment:
Do we need to test this -- seems to be not part of this test method and should be it's own test, if we don't have one yet.

Reply:
This was how the non-windowed test ended, so I just kept it. I can make it a separate test.